MBBSEmu/HostProcess/ExportedModules/Majorbbs.cs

```diff
@@ -7868,5 +7874,65 @@ private void txtvars_delegate()
 
             Registers.SetPointer(txtvarsReturnPointer);
         }
+
+        /// <summary>
+        ///     setmode - sets mode of open file
+        ///
+        ///     Signature: int setmode(int handle, int mode);
+        ///
+        /// FROM: fcntl.h:
+        /// * NOTE: Text is the default even if the given _O_TEXT bit is not on. */
+        /// #define _O_TEXT		0x4000	/* CR-LF in file becomes LF in memory. */
+        /// #define _O_BINARY	0x8000	/* Input and output is not translated. */
+        /// </summary>
+        private void setmode()
+        {
+            var fileHandle = GetParameter(0);
+            var fileMode = GetParameter(1);
+
+            if (!FilePointerDictionary.TryGetValue(fileHandle, out var fileStreamPointer))
+            {
+                // TODO sets errno to ???;
+                Registers.AX = 0xFFFF; // -1
+                return;
+            }
+
+            var fileStructPointer = Module.Memory.GetOrAllocateVariablePointer($"FILE_{fileStreamPointer.Name}-{FilePointerDictionary.Count}", FileStruct.Size);
+            var fileStruct = new FileStruct(Module.Memory.GetArray(fileStructPointer, FileStruct.Size));
+
+            switch (fileMode)
+            {
+                case 0x4000: //TEXT
```

Comment:
in theory a client could send a bitmask with both TEXT & BINARY set, but it's probably such a fringe case. Maybe we could throw an exception?

Reply:
Wouldn't the default: in the switch catch that exception? 
```
                default:
                    throw new Exception($"Unknown setmode: {fileMode}");
```